community/0006-governance.md

```diff
@@ -0,0 +1,37 @@
+
+# Motivation
+This document has been written with the explicit purpose of defining guidelines and expectations for project staff. It has been written with the idea of decentralizing power in mind, to help contribute to the goal of a truly open project.
+
+# Explanation
+
+## Governmental Structure
+![Government Structure](government-structure.png)
+\* For illustration purposes only, technical teams may vary
+
+### Teams
+A team is a group of staff members dedicated to solving a specific issue, maintaining a specific subproject, or overseeing a specific area. A team may be made up of any number of staff members, and a staff member may be on any number of teams. Teams may be community-focused, such as community moderators, or technically focused, such as the team of developers responsible for Quilt Loader. Team member responsibilities will vary and be detailed on a team-by-team basis.
+
+Each team will be initially constructed by the parent team responsible for that team. Further appointments will be voted upon by team members.
+
+#### Voting
+When a vote is called within a team, the majority wins. Members of the parent team may also vote if the voting team contains less than three members. In the event of a tie, members of the parent team may vote as well if they haven't already, with their votes being worth three votes.
+
+### Removal of Staff
+Any team member (up to and including administrative board members) may be removed by a majority vote of all impacted staff members. For members of the administrative, this is all staff members. For technical leads, this would be all technical staff members. For the QSL Module A lead, this would be all team members of QSL Module A. The vote must be initiated with a detailed call to action, describing why the staff member feels that the person should be removed from power. The accused will have one week to submit a detailed counterargument, at which point the vote will be called.
+
+### The Administrative Board
+The administrative board is the group of staff that holds ultimate power over the direction of the project. It is made up of three members so as to prevent deadlocks in the decision-making process. Members of the administrative board may be members of other teams within the project as well. Admins do not generally make use of their position unless a critical decision needs to be made whose team is unable to come to a consensus. In this case, the three admins will debate internally and make a decision. Acting in this way, the administrative board acts primarily as a tiebreaker.
+
+If there is an absence on the administrative board, each remaining board member may nominate a community member (potentially a non-staff member) for the position. Existing staff members may nominate themselves. The remaining staff members may then vote on the nominees, with administrative board member votes being worth three votes only in the event of a tie.
+
+# Drawbacks
+
```

Comment:
As a note to future readers -- this has been _Partially_ addressed in the Community Team RFC #7

Reply:
As an alternative _(and because I've already made the picture for this)_, one possible solution holding the Crown in [escrow](https://en.wikipedia.org/wiki/Escrow).

- Have a Discord account created solely for holding the Server Owner status within the server.
- Implement a system to have the three administrative board members hold access to a public-private key pair. The private key would be held in some system to allow only the confirmed board members to access it (this would need more thought).
- Use a software solution that implements the technical equivalent to the [two-man rule](https://en.wikipedia.org/wiki/Two-man_rule) (such as [cloudflare/redoctober](https://github.com/cloudflare/redoctober)), to encrypt the credentials of the Crown user with the three public keys such that any two of those board members can access the credentials to access the account.

If a system like this could be implemented, it would allow the Server Owner crown to remain offline always until at least two of the three administrative board members agrees to unlock it.

Is it overkill? Maybe. But it would put the power of the Server Owner crown in no one's hands.

_Addenum:_ if the previously-mentioned redoctober software is considered, note that it allows any number of owners of a secret, but a minimum number of them to access it. In this way, one could add more key pairs as secret owners for the Crown user, which are encrypted with other keys from the e.g. community team.

Attached herein (and hidden by default) is an infographic on how such a system (with an optional 4th key) would be structured:
<details>
<summary><strong>Crown Escrow</strong></summary>
<img src="https://user-images.githubusercontent.com/21304337/115559722-8572c080-a2e6-11eb-8136-5b03be7eef04.png">
</details>

Reply:
Escrow is a much more complicated process than is necessary here, but it may be revisited in the future. For now, the relatively indifferent keyholder is a good enough system.